src/openzaak/utils/inclusion.py

```diff
@@ -0,0 +1,222 @@
+# SPDX-License-Identifier: EUPL-1.2
+# Copyright (C) 2022 Dimpact
+from typing import List, Type
+
+from django.utils.module_loading import import_string
+
+from django_loose_fk.drf import FKOrURLField
+from django_loose_fk.loaders import FetchError
+from django_loose_fk.virtual_models import ProxyMixin
+from djangorestframework_camel_case.render import CamelCaseJSONRenderer
+from rest_framework.serializers import (
+    BaseSerializer,
+    HyperlinkedRelatedField,
+    Serializer,
+)
+from rest_framework_inclusions.core import InclusionLoader as _InclusionLoader, sort_key
+from rest_framework_inclusions.renderer import (
+    InclusionJSONRenderer as _InclusionJSONRenderer,
+)
+
+from connection_pooling.connections import use_connection_pool
+from openzaak.utils.serializer_fields import (
+    LooseFKHyperlinkedIdentityField,
+    LooseFKHyperlinkedRelatedField,
+)
+
+
+def get_component_name(serializer: Serializer) -> str:
+    return serializer.Meta.model._meta.app_label
+
+
+def get_resource_name(serializer: Serializer) -> str:
+    return serializer.Meta.model._meta.object_name
+
+
+def get_inclusion_key(serializer: Serializer) -> str:
+    component_label = get_component_name(serializer)
+    model_name = serializer.Meta.model._meta.model_name
+    return f"{component_label}:{model_name}"
+
+
+class InclusionLoader(_InclusionLoader):
+    # When doing inclusions, this indicates whether or not the entire path should
+    # be used to include nested resources, e.g.: `?include=resource1.resource2` vs `?include=resource2`
+    nested_inclusions_use_complete_path = True
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._seen_external = set()
+
+    def _has_been_seen_external(self, url: str) -> bool:
+        if url in self._seen_external:
+            return True
+        self._seen_external.add(url)
+        return False
+
+    def get_model_key(self, obj, serializer):
+        return get_inclusion_key(serializer)
+
+    def inclusions_dict(self, serializer):
+        entries = self._inclusions((), serializer, serializer.instance)
+        result = {}
+        for obj, inclusion_serializer in entries:
+            model_key = self.get_model_key(obj, inclusion_serializer)
+
+            # In case of external resources
+            if issubclass(type(obj), ProxyMixin):
```

Comment:
can't this be simplified as?

```suggestion
            if isinstance(obj, ProxyMixin):
```